assertsprocessor/metrics.go

```diff
@@ -0,0 +1,166 @@
+package assertsprocessor
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/collectors"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+)
+
+type metricHelper struct {
+	logger                *zap.Logger
+	config                *Config
+	prometheusRegistry    *prometheus.Registry
+	latencyHistogram      *prometheus.HistogramVec
+	attributeValueRegExps *map[string]*regexp.Regexp
+}
+
+func (p *metricHelper) compileSpanFilterRegexps() error {
+	p.logger.Info("consumer.Start compiling regexps")
+	for attName, matchExpString := range *p.config.AttributeExps {
+		compile, err := regexp.Compile(matchExpString)
```

Comment:
These regexes are validated in `config` too. Is this required here?

Reply:
> These regexes are validated in `config` too. Is this required here?
The compiled regexps created in Config can't be retained. Config doesn't have access to the rest of the components and it doesn't allow additional fields in it that are not intended to be part of configuration. So we just compile once again in the `consumer.Start` call